pkg/commands/compute/validate.go

```diff
@@ -89,9 +94,17 @@ func validate(path string) error {
 			}
 		}
 
-		err = f.Close()
+		if fileValidator != nil {
+			err = fileValidator(f)
+		}
+
+		closeErr := f.Close()
+
 		if err != nil {
-			return fmt.Errorf("error closing package: %w", err)
+			return err
+		}
+		if closeErr != nil {
+			return fmt.Errorf("error closing file: %w", closeErr)
```

Comment:
I think the `if err != nil` check can be moved inside of the `if fileValidator != nil` statement,  
and `closeErr` renamed to just `err`.

So something like:

```go
if fileValidator != nil {
	if err := fileValidator(f); err != nil {
		return err
	}
}

err = f.Close()
if err != nil {
	return fmt.Errorf("error closing file: %w", err)
}
```

Reply:
The reason for this is to avoid having to call `f.Close()` in two different places.
Are you saying that you want to omit the call, or that you are happy with the call being within the `fileValidator` error path?

Reply:
To be clear, the alternative I wanted to avoid was:

```go
if fileValidator != nil {
  if err := fileValidator(f); err != nil {
    f.Close()
    return err
  }
}
```

Reply:
OK that's understandable. I actually missed the fact that you would have had to close in two places.

So what would be the _practical_ outcome of not closing the file when we're in a CLI environment? My understanding is the moment the process stops the associated file descriptors will be automatically cleaned up.

If this was a long running server process, then I'd agree to always explicitly calling `f.Close()` but for a CLI I don't think it matters as much.

Also from a user experience, I personally don't feel it's a useful error to report back as a user can't do anything about it if it did fail so maybe if we do want to close the file we instead use `defer` and at just report it into the error log using `c.Globals.ErrLog`.